### DISCUSSION TITLE
Make `firejail --appimage` go through proxychains

### DISCUSSION THREAD

**Discussion by tredondo** [netblue30/firejail#4701] [Q&A] [answered]
I'm trying to get [this LosslessCut appimage](https://github.com/mifi/lossless-cut/releases/tag/v3.39.0) to work with firejail and proxychains (through a SOCKS5 proxy). Found this post on [using `--env`](https://github.com/netblue30/firejail/issues/1160):

    firejail --noprofile --env="LD_PRELOAD=libproxychains.so.4" --appimage XnView_MP.0.99.1.AppImage

That works with some appimages (e.g. [XnView](https://download.xnview.com/XnView_MP.glibc2.18-x86_64.AppImage) above), but doesn't work with others (apparently Electron apps?). The command exits immediately:

```
$ firejail --noprofile --env="LD_PRELOAD=libproxychains.so.4" --appimage LosslessCut-linux_3.39.0.AppImage 
Parent pid 611120, child pid 611123

** Warning: dropping all Linux capabilities and setting NO_NEW_PRIVS prctl **

Mounting appimage type 2
Child process initialized in 68.82 ms
[proxychains] DLL init: proxychains-ng 4.14
[proxychains] DLL init: proxychains-ng 4.14
[proxychains] DLL init: proxychains-ng 4.14
[proxychains] DLL init: proxychains-ng 4.14

Parent is shutting down, bye...
AppImage detached
```

If I run just `proxychains firejail --appimage --profile=no <AppImage>`, that launches the appimage, but proxychains doesn't intercept any network calls. That happens [for any binary launched by firejail](https://github.com/haad/proxychains/issues/124), actually.

**Comment by rusty-snake** [comment, marked as answer]
Does `LD_PRELOAD=libproxychains.so.4 ./LosslessCut-linux_3.39.0.AppImage` work? (i.e. is this caused by firejail or by electron?).

**Reply by tredondo** [reply]
No, I get "[proxychains] DLL init: proxychains-ng 4.14" a few times, then `Trace/breakpoint trap`.

**Reply by rusty-snake** [reply]
Yeah, that's it, electron/chrom*/firefox does not like to LD_PRELOAD libraries into it.
